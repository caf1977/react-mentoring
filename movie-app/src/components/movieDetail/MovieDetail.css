.movie-detail {
    padding: 10px 10px 20px 10px;
    display: flex;
    background-color: #232323;
}

.movie-detail img {
    height: 300px;
    width: 250px;
}

.movie-detail-info {
    font-size: 14px;
    font-weight: lighter;
    color: white;
    padding: 15px;
    display: flex;
    flex-direction: column;
    padding-left: 30px;
    padding-top: 0;
    width: 60%;
}

.movie-detail-name {
    font-size: 24px;
    font-weight: lighter;
    color: white;
    text-transform: uppercase;
}

.movie-detail-rating {
    font-weight: bold;
    border: 2px solid white;
    border-radius: 50%;
    width: 40px;
    height: 30px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-left: 20px;
}

.movie-detail-header {
    display: flex;
}

.movie-detail-genres {
    font-size: 10px;
}

.movie-detail-middle {
    display: flex;
}

.movie-detail-middle h2 {
    color: #f65261;
    margin-top: 15px;
    margin-bottom: 5px;
    margin-right: 30px;
    font-size: 14px;
    font-weight: lighter;
}