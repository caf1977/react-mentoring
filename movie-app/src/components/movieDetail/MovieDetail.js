import './MovieDetail.css'

const MovieDetail = ({movieInfo}) => {
    return (
        <div className="movie-detail">
            <img src={movieInfo.poster_path} alt={movieInfo.title}/>
            <div className="movie-detail-info">
                <div className="movie-detail-header">
                    <div className="movie-detail-name">{movieInfo.title}</div>
                    <div className="movie-detail-rating">{movieInfo.vote_average}</div>
                </div>
                <div className="movie-detail-genres">{movieInfo.genres.join(",")}</div>
                <div className="movie-detail-middle">
                    <h2>{movieInfo.release_date}</h2>
                    <h2>{movieInfo.runtime}</h2>
                </div>
                <p>{movieInfo.overview}</p>
            </div>
        </div>
    )
}

export default MovieDetail