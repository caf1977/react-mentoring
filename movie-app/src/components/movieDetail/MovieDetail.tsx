import React, { FC } from 'react';
import './MovieDetail.css'
import { MovieInfo } from '../../types/MovieInfo';

interface IMovieDetail {
    movieInfo: MovieInfo;
}

const MovieDetail: FC<IMovieDetail> = ({movieInfo}) => {
    return (
        <div className="movie-detail">
            <img src={movieInfo.poster_path} alt={movieInfo.title}/>
            <div className="movie-detail-info">
                <div className="movie-detail-header">
                    <div className="movie-detail-name">{movieInfo.title}</div>
                    <div className="movie-detail-rating">{movieInfo.vote_average}</div>
                </div>
                <div className="movie-detail-genres">{movieInfo.genres?.join(",")}</div>
                <div className="movie-detail-middle">
                    <h2>{movieInfo.release_date}</h2>
                    <h2>{movieInfo.runtime}</h2>
                </div>
                <p>{movieInfo.overview}</p>
            </div>
        </div>
    )
}

export default MovieDetail