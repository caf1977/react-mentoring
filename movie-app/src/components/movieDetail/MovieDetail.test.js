import {render, screen} from "@testing-library/react";
import MovieDetail from "./MovieDetail";

describe("MovieDetail component tests", () => {

    const movie = {
        imageUrl: "https://upload.wikimedia.org/wikipedia/en/0/00/Spider-Man_No_Way_Home_poster.jpg",
        movieName: "Spider Man",
        releaseYear: 2020,
        rating: "PG",
        duration: "120 minutes",
        description: "Peter Parker returns home to live with his Aunt May and mentor Tony Stark after his debut as Spider-Man in Captain America: Civil War.",
    };

    it("renders movie detail info", () => {
        render(<MovieDetail movieInfo={movie}/>);
        const image = screen.getByAltText("Spider Man");
        const name = screen.getByText("Spider Man");
        const releaseYear = screen.getByText("2020");
        const rating = screen.getByText("PG");
        const duration = screen.getByText("120 minutes");
        const description = screen.getByText("Peter Parker returns home to live with his Aunt May and mentor Tony Stark after his debut as Spider-Man in Captain America: Civil War.");
        
        expect(image.src).toContain("https://upload.wikimedia.org/wikipedia/en/0/00/Spider-Man_No_Way_Home_poster.jpg");
        expect(name).toBeInTheDocument();
        expect(releaseYear).toBeInTheDocument();
        expect(rating).toBeInTheDocument();
        expect(duration).toBeInTheDocument();
        expect(description).toBeInTheDocument();
    })

    it("renders snapshot", () => {
        const { asFragment } = render(<MovieDetail movieInfo={movie}/>);

        expect(asFragment(<MovieDetail movieInfo={movie}/>)).toMatchSnapshot();
    })
})