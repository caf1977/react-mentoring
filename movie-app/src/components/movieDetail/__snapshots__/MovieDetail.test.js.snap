// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail component tests renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-detail"
  >
    <img
      alt="Spider Man"
      src="https://upload.wikimedia.org/wikipedia/en/0/00/Spider-Man_No_Way_Home_poster.jpg"
    />
    <div
      class="movie-detail-info"
    >
      <p>
        <b>
          Name:
        </b>
         Spider Man
      </p>
      <p>
        <b>
          Release year:
        </b>
         2020
      </p>
      <p>
        <b>
          Rating:
        </b>
         PG
      </p>
      <p>
        <b>
          Duration:
        </b>
         120 minutes
      </p>
      <p>
        <b>
          Description:
        </b>
         Peter Parker returns home to live with his Aunt May and mentor Tony Stark after his debut as Spider-Man in Captain America: Civil War.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
